pkg/epp/scheduling/plugins/filter/has_capacity_filter.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/config"
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/plugins"
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/types"
+)
+
+// compile-time type validation
+var _ plugins.Filter = &HasCapacityFilter{}
+
+// NewHasCapacityFilter returns a new HasCapacityFilter.
+func NewHasCapacityFilter() *HasCapacityFilter {
+	return &HasCapacityFilter{
+		queueThreshold:   config.Conf.QueueThresholdCritical,
+		kvCacheThreshold: config.Conf.KVCacheThreshold,
+	}
+}
+
+// HasCapacityFilter filters only pods that has capacity for sheddable requests.
+type HasCapacityFilter struct {
+	queueThreshold   int
+	kvCacheThreshold float64
+}
+
+// Name returns the name of the filter.
+func (f *HasCapacityFilter) Name() string {
+	return "has-capacity"
+}
+
+// Filter filters out pods that doesn't meet the filter criteria.
+func (f *HasCapacityFilter) Filter(ctx *types.SchedulingContext, pods []types.Pod) []types.Pod {
+	filteredPods := []types.Pod{}
+
```

Comment:
I'm reading this now again (reviewing myself lol).
@liu-cong does it make sense to get rid of Sheddable filter and put the following condition?
sheddable request doesn't do anything other than that..
anyway, if you agree with me, please let's defer to next PR since this is a large enough and I don't want to include more changes unless it's a must.
```go
if ctx.Req.Critical == true {
    return pods
}
```

Reply:
Yeah it makes sense to combine the two, perhaps call it `HasCapacityForSheddableRequestFilter`. Feel free to refactor or defer it.